packages/compass/src/main/index.ts

```diff
@@ -21,34 +25,62 @@ if (process.env.APP_ENV === 'webdriverio') {
 // @ts-expect-error setVersion is not a public method
 app.setVersion(process.env.HADRON_APP_VERSION);
 
-const importExportOptions = {
-  // TODO(COMPASS-6070): Proper command line parsing
-  exportConnections: process.argv.find(s => s.startsWith('--export-connections='))?.slice(21),
-  importConnections: process.argv.find(s => s.startsWith('--import-connections='))?.slice(21),
-  passphrase: process.argv.find(s => s.startsWith('--passphrase='))?.slice(13),
-  // TODO(COMPASS-6066): Set removeSecrets: true if protectConnectionStrings is set.
-  removeSecrets: false,
-  trackingProps: { context: 'CLI' },
-};
-
 void main();
 
 async function main(): Promise<void> {
+  const globalPreferences = await parseAndValidateGlobalPreferences();
+  const { preferenceParseErrors } = globalPreferences;
+  const preferenceParseErrorsString = preferenceParseErrors.join('\n');
+  if (globalPreferences.cli.version) {
+    process.stdout.write(`${app.getName()} ${app.getVersion()}\n`)
+    return app.exit(0);
+  }
+
+  if (globalPreferences.cli.help) {
+    process.stdout.write(getHelpText());
+    return app.exit(0);
+  }
+
+  if (preferenceParseErrors.length > 0) {
+    process.stderr.write(chalk.yellow(preferenceParseErrorsString) + '\n');
+    process.stderr.write('Use --ignore-additional-command-line-flags to allow passing additional options to Chromium/Electron\n');
+  }
```

Comment:
If we age getting strict with this maybe would make sense to allow at least some flags, like disable-gpu, that we recommended a few times as a workaround for visual glitches on macos and ubuntu crashes?

Reply:
Yeah, I’ve also gone back and forth a bit on this. There seems to be no exhaustive list of Electron and Chromium flags, let alone a way to programmatically access them. Do you know of other flags that we might want to allow here? `--disable-gpu` and `--no-sandbox` are the only obvious candidates for me right now.

Reply:
Yeah, this is super fair, I think all autogenerated lists for switches I saw were building it from chromium source, but I don't think it's really worth it for us. I think `disable-gpu` and `no-sandbox` are very good candidates and nothing else comes to mind, let's do them for now maybe and see if there is ever any need to expand when we release this?

Reply:
Sounds good! Done in 13113284239aa2ff6b51cf5df00ea57a7f3d5bfd :+1: